media-libs/opusfile/files/http-fix-compilation-with-LibreSSL.patch

```diff
@@ -0,0 +1,114 @@
+From 8486e83cdba2d8c537edeacfc33eabb7ebf9f0e5 Mon Sep 17 00:00:00 2001
```

Comment:
Was this patch applied upstream?

Reply:
Not yet. Here is the issue I created: https://gitlab.xiph.org/xiph/opusfile/issues/2327
and the same patch: https://gitlab.xiph.org/xiph/opusfile/merge_requests/2

The reviewer said that he likes the nginx approach more — to redefine OPENSSL_VERSION_NUMBER to 0x1000107fL if LibreSSL is used. I will make such a patch for the upstream shortly.

Reply:
I sent another (alternative) patch for the upstream, and it has been applied:
https://git.xiph.org/?p=opusfile.git;a=commitdiff;h=d59a17ec1e58

Reply:
So I have updated this PR with the patch applied upstream.

Reply:
Thanks. However, I would really prefer if you added appropriate `>=` conditional to use new API with new LibreSSL. It's not nice to force deprecated functions on LibreSSL users.

Reply:
I agree, the additional conditionals for the new LibreSSL would be nicer, something like:
`# if (OPENSSL_VERSION_NUMBER<0x10100000L&&LIBRESSL_VERSION_NUMBER<0x2080000fL)`

But I think this should be applied upstream first. Therefore another patch is needed, as the above -mentioned is already applied. Maybe it worth to apply the patch already applied upstream, to close the bug in Gentoo? I'll send the new patch upstream shortly. And then I'll update the patch for the ebuild with the new conditionals as soon as it will be accepted upstream, right?

Reply:
I don't think it would harm to wait a few days, and possibly apply a combined patch in Gentoo. Please let me know if upstream doesn't apply it in reasonable time.